File: src/test/java/com/shurikat/utils/SortAssert.java
package com.shurikat.utils;

import org.junit.Assert;

/**
 * @author Alex Ivchenko
 */
public class SortAssert extends Assert {
    public static <T extends Comparable<? super T>> void assertSorted(T[] a) {
        for (int i = 1; i < a.length; ++i) {
            if (a[i].compareTo(a[i - 1]) < 0) {
                throw new AssertionError("not sorted");
            }
        }
    }
}

File: src/test/kotlin/com/shurikat/utils/SortAssert.kt
package com.shurikat.utils

/**
 * @author Alex Ivchenko
 */
fun <T : Comparable<T>> assertSorted(a: Array<T>) {
    (1 until a.size)
            .filter { a[it] < a[it - 1] }
            .forEach { throw AssertionError("not sorted ${it - 1}, $it: (${a[it - 1]}, ${a[it]})") }
}